src/language.cpp

```diff
@@ -35,7 +35,31 @@ TextString::TextString()
 void TextString::Clear()
 {
 	this->name = nullptr;
-	std::fill_n(this->languages, lengthof(this->languages), nullptr);
+	std::fill_n(this->languages, lengthof(this->languages), std::vector<const char*>());
+}
+
+/**
+ * Get the string in the currently selected language.
+ * @param amount The amount for which to return the appropriate plural form.
```

Comment:
`amount` expresses an imprecise number in many cases, a quantity that has no exactly known value ("a substantial amount of work"). Here you used exactly known values to find the plural form.

Maybe "value", "count", or "number" or so?

Also only this line talks about plural rather than line 42 and possibly 44 which is about promises made by the function.

"for which" is not proper English. I am guessing it's a literal translation of some German phrase.
You can simplify it like "The value to use for selecting the appropriate plural form."  No need to have "to return" in there. Line 44 is about returned values :)